components/server/src/common.rs

```diff
@@ -514,17 +516,34 @@ impl EnginesResourceInfo {
     ) {
         let mut compaction_pending_bytes = [0; DATA_CFS.len()];
         let mut soft_pending_compaction_bytes_limit = [0; DATA_CFS.len()];
+        // level0 file number ratio within [compaction trigger, slowdown trigger].
+        let mut level0_ratio = [0.0f32; DATA_CFS.len()];
 
         let mut fetch_engine_cf = |engine: &RocksEngine, cf: &str| {
             if let Ok(cf_opts) = engine.get_options_cf(cf) {
+                let offset = data_cf_offset(cf);
                 if let Ok(Some(b)) = engine.get_cf_pending_compaction_bytes(cf) {
-                    let offset = data_cf_offset(cf);
                     compaction_pending_bytes[offset] += b;
                     soft_pending_compaction_bytes_limit[offset] = cmp::max(
                         cf_opts.get_soft_pending_compaction_bytes_limit(),
                         soft_pending_compaction_bytes_limit[offset],
                     );
                 }
+                if let Ok(Some(n)) = engine.get_cf_num_files_at_level(cf, 0) {
+                    let level0 = n as f32;
+                    let slowdown_trigger = cf_opts.get_level_zero_slowdown_writes_trigger() as f32;
+                    let compaction_trigger =
+                        cf_opts.get_level_zero_file_num_compaction_trigger() as f32;
+                    let ratio = if slowdown_trigger > compaction_trigger {
+                        (level0 - compaction_trigger) / (slowdown_trigger - compaction_trigger)
+                    } else {
+                        1.0
```

Comment:
Is it mean if slowdown_trigger <= compaction_trigger, we disable the limiter?

Reply:
Yes.